bin/check_deadlock.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+#config
+user=root
+passwd=
+fe_ip=127.0.0.1
+http_port=8030
+timeout=30
+max_timeout_times=5
+fe_bin_path=/home/disk1/sr/app/starrocks-2.3.3/fe-5963b718-fdc4-42fc-a83c-5256cdcef805/bin
+jstack_path=/home/disk1/sr/app/starrocks-2.3.3/fe-5963b718-fdc4-42fc-a83c-5256cdcef805
+
+
+timeout_times=0
+interval=60
+while true
+do
+    # call show proc '/statistic' to check db deadlock
+    start_time=$(date +%s)
+    curl -u ${user}:${passwd} http://${fe_ip}:${http_port}/api/show_proc?path=/statistic --max-time ${timeout}
+    succ=$?
+    end_time=$(date +%s)
+
+    # check result, if return failed and time used is at least ${timeout}, increase timeout_times
+    if [ ${succ} -ne 0 ]; then
+	if [ $(($end_time - $start_time)) -ge ${timeout} ]; then
+       	    timeout_times=$(($timeout_times + 1))
+            # reduce interval, so that we can detect deadlock ASAP
+            interval=10
+            echo -e "\ntimeout, time used: $(($end_time - $start_time))"
+        else
+            timeout_times=0
+            echo -e "\ncheck failed, time used: $(($end_time - $start_time))"
+        fi
+    else
+        timeout_times=0
+        interval=60
+        echo -e "\ncheck successfully, time used: $(($end_time - $start_time))"
+    fi
+
+    # check ${timeout_times}, if ${timeout_times} >= max_timeout_times print jstack and stop fe
+    # we will not start fe, Because the process is hosted by supervisor
```

Comment:
suppose this script will be running on production, rarely the root password is empty, and it is not good practice to write password plaintext in this script, which makes this script useless.